models/repo.go

```diff
@@ -470,6 +470,12 @@ func MigrateRepository(u *User, name, desc string, private, mirror bool, url str
 		return repo, fmt.Errorf("create update hook: %v", err)
 	}
 
+	// Check if repository is empty.
+	_, _, err = com.ExecCmdDir(repoPath, "git", "rev-parse", "--verify", "--quiet", "HEAD")
```

Comment:
So, we these alternatives:

    git log -1
    fatal: bad default revision 'HEAD'

    git rev-parse --verify HEAD
    fatal: Needed a single revision

If you think checking stderr is more reliable than checking err, I could update the commit to do something like this instead:

    _, stderr, err := com.ExecCmdDir(repoPath, "git", "log", "-1")
    if err != nil && strings.Contains(stderr, "fatal:") { ... }

Let me know what you think.